src/validator/CreditCardVendorsReader.java

```
@@ -0,0 +1,58 @@
+package validator;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class CreditCardVendorsReader implements CreditCardVendorReading {
+	@Override
+	public ArrayList<CreditCardVendor> returnVendorArray() {
+		ArrayList<CreditCardVendor> list = new ArrayList<CreditCardVendor>();
+		
+		try(BufferedReader br = new BufferedReader(new FileReader("Base_of_credit_card_vendors.txt"))) {
+		    for(String line; (line = br.readLine()) != null; ) {
+		        String[] values = line.split(" ");
+		        
+		        String vendorName = values[0];
+		        ArrayList<Integer> vendorNumbersOfChars = new ArrayList<Integer>();
+		        ArrayList<String> vendorMasks = new ArrayList<String>();
+		        
+		        parseLengths(values[1], vendorNumbersOfChars);
+		        parseMasks(values[2], vendorMasks);
+		        
+		        list.add(new CreditCardVendor(vendorName,vendorNumbersOfChars,vendorMasks));
+		        
+		    }
+		} catch (FileNotFoundException e) {
+			e.printStackTrace();
```

system cannot operate without having loaded the vendors. fail fast to abort rather than just logging the exception and continuing execution. FileNotFound is self explanatory, so I'd be fine with `@lombok.SneakyThrows`